### PR TITLE
cadams/fix parser tests

### PR DESCRIPTION
**Pull request by chris-chen-creates**
Opening PR for test fixes and test additions.

### CODE REVIEW

File: src/Parser.ts [hunk 1]
```diff
@@ -17,35 +17,44 @@ class ParserError extends Error {}
...
 
-  constructor(
-    public tokens: Token[],
-  ) {}
+  constructor(public tokens: Token[]) {}
 
   parse(): Expression {
     const expr = this.expression()
+    if (this.isAtEnd() == false) {
+      throw new ParserError('unparsed tokens')
```
Comment (angles-n-daemons) [inline src/Parser.ts]: can you make this error a little more descriptive?

File: src/Parser.ts [hunk 2]
```diff
@@ -94,20 +106,31 @@ export class Parser {
...
 
   advance(): Token {
-    this.index++
+    if (!this.isAtEnd()) this.index++
     return this.previous()
   }
 
   previous(): Token {
-    return this.tokens[this.index]
+    if (this.tokens[this.index - 1] == undefined) {
```
Comment (angles-n-daemons) [inline src/Parser.ts]: let's make this more explicit -

```
if (this.index < 1) { ...
```

**Review by angles-n-daemons — CHANGES_REQUESTED** [review]
Couple of changes requested

File: src/Parser.ts (hunk 1 above)
Comment (chris-chen-creates) [reply src/Parser.ts]: Got that updated with a more descriptive error.

File: src/Parser.ts (hunk 2 above)
Comment (chris-chen-creates) [reply src/Parser.ts]: Simplified this logic to be more explicit.

**Comment by chris-chen-creates** [comment]
This branch is a bit behind since I did some additional work out of cadams/expression-and-grammar-additions.

**Review by angles-n-daemons — APPROVED** [review]

**Pull request merged**